src/v/archival/ntp_archiver_service.cc

```diff
@@ -2026,15 +2026,29 @@ ss::future<ntp_archiver::batch_result> ntp_archiver::wait_all_scheduled_uploads(
       .compacted_upload_result = compacted_result};
 }
 
+model::offset ntp_archiver::max_uploadable_offset() const {
+    // We impose an additional (LSO) constraint on the uploadable offset to
+    // as we need to have a complete index of aborted transactions if any
+    // before we can upload a segment.
+    return std::min(
+      model::prev_offset(_parent.last_stable_offset()),
+      _parent.committed_offset());
+}
```

Comment:
Is it ever possible that `_parent.last_stable_offset()` and `_parent.committed_offset()` are both 0? 

Then this method will return int64_t min value because of `prev_offset`, and later on `max_stable_offset` becomes min value + 1. 

This still shouldn't cause any issues, just might be hard to understand in logs. I don't see an alternative though so this is fine.

Reply:
I did scratch my head a bit too. Theoretically it is impossible for both to be 0 because comitted_offset always lags by a value >= 1 than LSO.

Found somewhat surprising that next_offset of any negative value is just 0

Reply:
`model::prev_offset` from the `model::offet(0)` is `model::offset{}` so this functions are symmetrical